src/basics.md

```diff
@@ -748,6 +749,63 @@ fn run() -> Result<()> {
 
 ```
 
+[ex-sha-digest]: #ex-sha-digest
+<a name="ex-sha-digest"></a>
+## Calculate the SHA-256 digest of a file
+
+[![ring-badge]][ring] [![data-encoding-badge]][data-encoding] [![cat-filesystem-badge]][cat-filesystem]
+
+Writes some data to a file, then calculates the SHA-256 digest of
+the file's contents using [`digest::Context`].
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+
```

Comment:
please add one more `#` on the begining of line here so we don't have weird empty lines in final output